### ISSUE TITLE
Bug: Error message "unsupported url"

### ISSUE THREAD

**Issue by tordans** [bryceco/GoMap#212]
I wrote a longer commit message today and received a weird error message 

> Unable to upload changes
> Unsupported URL
> <OK>

Once I shortened the message by removing everything including the "// …", the commit went through. 

I am pretty sure this error message is not right here. Cause I did not change anything URL related, but the commit went through.
- Maybe the "message too long" error has the wrong message?
- Maybe the "//" in the commit message causes a weird error?

<img src="https://user-images.githubusercontent.com/111561/56148525-c70b5b00-5faa-11e9-8ff4-371b4ea86e6f.png" width=350>

**Comment by wtimme** [comment]
Thanks for your report! I can reproduce the error on my end.

In order to reproduce this, just one `/` is enough. I'll have a look.

**Comment by bryceco** [comment]
You might also check if there’s a 255 character limit and what happens.

**Comment by wtimme** [comment]
I've identified at least two ways that might cause the behaviour:

1. If the API URL is changed by the user and does not contain the suffix `/`, it will [add two `//`][1] to the string.
2. The login view controller uses the values from the text fields in `viewWillDisappear:` and saves them _unverified_ to the `KeyChain`. To add to that, the `username` field has auto-correction enabled. These two issues combined have resulted in an "auto-corrected" username when the view controller disappears, but iOS decides last-minute to apply the suggested auto-correction.

[1]: https://github.com/bryceco/GoMap/blob/master/src/Shared/OsmMapData.m#L126

**Comment by bryceco** [comment]
1 is obviously a bug.
2 isn’t great, but assuming people press Verify instead of Back it isn’t a problem. 

Neither of those seems to be the cause of the error Tobias saw, don’t you agree?

**Comment by wtimme** [comment]
Saving unverified credentials causes the alert that is visible in the screenshot from the report.

## Steps to reproduce

1. Open the Settings and tap "OpenStreetMap Login"
2. Enter credentials that are invalid, e. g. "foo" for the username and "bar" for the password
3. Tap "Verify". After the request is done, you'll get an error message "Bad login - Not found"
4. Go back to the map
5. Do some changes, e. g. add a new node
6. Tap the "Upload" bar button item
7. Tap "Commit"

**Tip:** For testing, you can use the development instance at `https://master.apis.dev.openstreetmap.org/`.

---

Sniffing the network traffic (e. g. using "Charles Proxy"), I could tell that the app was trying to use the unverified (and invalid) credentials when trying to create the new changeset.
The server responds with "Couldn't authenticate you", but the app treats _any_ string as a changeset ID (and therefore a success) and continues with the next request - uploading the changeset XML.

Wth #215, I've added a check for the changeset ID that we get from the server. If it contained any other character than digits, it is treated as an error message. Here's what I get now:

![couldnt-authenticate-you](https://user-images.githubusercontent.com/1681085/56165609-ee6e2200-5fc2-11e9-80e2-2ee57401b5a9.png)

**Comment by tordans** [comment]
> Which version are you on?

v1.6.2, the current test flight

**Comment by tordans** [comment]
> Steps to reproduce
> …

I was mapping a lot the last few days with the published version and the test flight version. I did not touch my credentials for ages. So I wonder if it was a weird network-traffic issue? But I am quite sure I retried (tried to submit it again) to check for that.

The only place in the app I could find the error message is https://github.com/bryceco/GoMap/blob/3fc1e0d170a625416caf1290fbecd9d0ab8ac937/src/Shared/AerialList.m#L419 which does not seem to have anything to do with uploading at all, right?

However, I only saw it once, so I thinks it very OK to just see if it comes up again after your improvements!
